<launch>

    <arg name="drone_name" default="drone2" />

    <param name="/use_sim_time" type="bool" value="false"/>

    <rosparam command="load" file="$(find rnw_ros)/config/swarm_rnw_config.yaml"/>

    <!--  Setup Gamepad  -->

    <node pkg="joy" type="joy_node" name="joy_node" output="screen">
        <param name="~autorepeat_rate" value="0" />
    </node>

    <group>
        <remap from="joy" to="joy_hz"/>
        <node pkg="joy" type="joy_node" name="joy_cont" output="screen">
            <param name="~autorepeat_rate" value="30" />
        </node>
    </group>

    <node pkg="rnw_ros" type="gamepad_trigger_node" name="gamepad" output="screen">
        <remap from="~RB" to="/abort"/>
    </node>

    <!--  Setup UWB  -->

    <node pkg="mocap_optitrack" type="mocap_node" name="mocap_node" respawn="false" launch-prefix="" required="true" output="log">
        <rosparam file="$(find mocap_optitrack)/config/mocap.yaml" command="load" />
    </node>

    <node pkg="mocap2odom" type="mocap2odom" name="mocap2odom" output="screen" >
        <param name="publish_uav_odom" type="bool" value="true"/>
        <rosparam file="$(find mocap2odom)/config/calib.yaml" command="load" />
    </node>

    <node pkg="uwb_transceiver" type="uwb_transceiver_node" name="uwb_master_$(arg drone_name)" output="screen">
        <rosparam command="load" file="$(find uwb_transceiver)/config/config.yaml" />
        <param name="serial_port" value="/dev/tty.uwb.$(arg drone_name)"/>
        <remap from="~odom" to="/$(arg drone_name)/odom"/>
        <remap from="~position_cmd" to="/$(arg drone_name)/position_cmd"/>
        <remap from="~n3ctrl" to="/$(arg drone_name)/state"/>
        <remap from="~trigger" to="/$(arg drone_name)/trigger"/>
        <param name="role" type="int" value="1"/> <!--master-->
        <param name="debug/m_debug_packet_info" type="int" value="0"/>
    </node>

    <node pkg="rnw_ros" type="traj2cmd_node" name="traj2cmd_$(arg drone_name)" output="screen" required="true">
        <remap from="~traj" to="/$(arg drone_name)/traj"/>
        <remap from="~odom" to="/$(arg drone_name)/odom"/>
        <remap from="~position_cmd" to="/$(arg drone_name)/position_cmd"/>
    </node>

    <!--  RnW  -->

    <node pkg="rnw_ros" type="pub_cone_state_node" name="pub_cone_state_node" required="true" output="log">
        <param name="cut_euler_velocity" type="bool" value="true"/>
        <param name="max_euler_velocity" value="10"/>
    </node>

    <node pkg="rnw_ros" type="individual_flight_test_node" name="individual_flight_test_node" required="true" output="screen">
        <param name="drone_name" type="string" value="$(arg drone_name)"/>
    </node>

    <!--  Visualization  -->

    <node pkg="rnw_ros" type="visualize_pos_cmd_node" name="pos_cmd_viz_$(arg drone_name)" output="screen" required="true">
        <remap from="~position_cmd" to="/$(arg drone_name)/position_cmd"/>
    </node>

    <node pkg="rnw_ros" type="visualize_traj_node" name="traj_viz_$(arg drone_name)" output="screen" required="true">
        <param name="lift_dt" type="double" value="0.05"/>
        <param name="length_g" type="double" value="0.5"/>
        <param name="clear_after_n_sec" type="double" value="0"/>
        <remap from="~poly_traj" to="/$(arg drone_name)/traj"/>
    </node>

    <node pkg="rnw_ros" type="odom2tf_node" name="$(arg drone_name)_tf" output="screen">
        <remap from="~odom" to="/$(arg drone_name)/odom"/>
        <param name="name" value="odom_$(arg drone_name)" />
    </node>

    <node pkg="rnw_ros" type="visualize_cone_node" name="visualize_cone_node" output="screen" required="true">
        <param name="cone_color_r" type="double" value="0"/>
        <param name="cone_color_g" type="double" value="1"/>
        <param name="cone_color_b" type="double" value="0"/>
    </node>

<!--    <node name="rviz_traj" pkg="rviz" type="rviz" output="log" required="false" args="-d $(find rnw_ros)/config/swarm.rviz" />-->

<!--    <node name="record" pkg="rosbag" type="record" output="screen" args="-a -x '(.*)visual(.*)|/(.*)viz(.*)'" />-->

<!--    <node pkg="rqt_multiplot" type="rqt_multiplot" name="rqt_multiplot" output="log" required="true"-->
<!--          args="&#45;&#45;multiplot-config $(find rnw_ros)/config/rqt_multiplot_tuning_drone2.xml &#45;&#45;multiplot-run-all"/>-->

<!--    <node pkg="rqt_reconfigure" type="rqt_reconfigure" name="rqt_reconfigure"/>-->

</launch>
